src/CTA.Rules.Metrics/Models/BuildErrorMetric.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Newtonsoft.Json;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace CTA.Rules.Metrics
+{
+    public class BuildErrorMetric : CTAMetric
+    {
+        // Match pattern for compiler error codes
+        // example: "CS1234: "
+        private const string BuildErrorCodePattern = "CS[0-9]{4}"; 
+
+        [JsonProperty("metricName", Order = 10)]
+        public string MetricName => "BuildError";
+
+        [JsonProperty("buildErrorCode", Order = 11)]
+        public string BuildErrorCode { get; set; }
+
+        [JsonProperty("buildError", Order = 12)]
+        public string BuildError { get; set; }
+
+        [JsonProperty("count", Order = 20)]
+        public int Count { get; set; }
+
+        [JsonProperty("solutionPath", Order = 30)]
+        public string SolutionPathHash { get; set; }
+
+        [JsonProperty("projectGuid", Order = 40)]
+        public string ProjectGuid { get; set; }
+
+        public BuildErrorMetric(MetricsContext context, string buildError, int count, string projectPath)
+        {
+            BuildErrorCode = ExtractBuildErrorCode(buildError);
+            BuildError = buildError;
+            Count = count;
+            SolutionPathHash = context.SolutionPathHash;
+            ProjectGuid = context.ProjectGuidMap.GetValueOrDefault(projectPath, "N/A");
+        }
+
+        private string ExtractBuildErrorCode(string buildError)
+        {
+            var pattern = new Regex(BuildErrorCodePattern);
+            var matches = pattern.Matches(buildError);
+            var errorCode = matches.FirstOrDefault(m => buildError.StartsWith(m.Value))?.Value;
+
+            return errorCode ?? string.Empty;
```

Comment:
if it doesn't start with "cs", we can still keep it ? what's wrong with that..

Reply:
For `BuildErrorMetric.BuildErrorCode`, we are parsing an error message string to get the code.  Based on [the docs for msbuild](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/), we try parsing out the patten `CS####`. 

If we get a build error message that does not contain an error code with this pattern, we still log the error, it just won't have a code associated with it.